src/tree.rs

```diff
@@ -436,6 +438,63 @@ impl<T> Tree<T> {
     }
 
     ///
+    /// Swaps two `Node`s including their children given their `NodeId`s.
+    ///
+    /// Returns an empty `Result` containing a `NodeIdError` if one occurred on either provided `Id`.
+    ///
+    /// Notes:
+    ///     - Both `NodeId`s are still valid after this process and are not swapped.
+    ///     - Keeps the positions of the `Node`s in their parents children collection.
+    ///
+    /// ``
```

Comment:
Yep, that makes sense to me. Thanks for drawing that up.

And yeah, writing tests for this won't be pretty.  If you want we can keep this off of Master for a bit too make sure we get the tests ironed out first. Up to you.

Let me know if you want me to set up a dev branch for this.